### PR TITLE
updates nxos_facts with more fact collections

### PR DESCRIPTION
**Pull request by privateip**
- adds support for std network facts
- adds support for default facts subset
- adds support for config facts subset
- adds support for interface facts subset
- adds support for hardware facts subset
- maintains backwards capabilitity with 2.1 facts module

Tested on NXOS 7.3(0)D1(1)

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @privateip. This module is going into community review.

We notice that you are one of the maintainers (@privateip, @GGabriele, @gundalow) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

File: network/nxos/nxos_facts.py [hunk 1]
```diff
@@ -16,253 +16,521 @@
...
-        body = [json.loads(response[0])]
-    return body
-
-
-def execute_show(cmds, module, command_type=None):
+      - When supplied, this argument will restrict the facts collected
+        to a given subset.  Possible values for this argument include
+        all, hardware, config, legacy, and interfaces.  Can specify a
+        list of values to include a larger subset.  Values can also be used
+        with an initial M(!) to specify that a specific subset should
```
Comment (gundalow) [inline network/nxos/nxos_facts.py]: `M(!)` will be formatted as a module link, to '!' 

Should this be `C(M(!))` of you want that string to appear?

Comment (gundalow) [reply network/nxos/nxos_facts.py]: Or maybe put  an example in,  `!Hardwate`

**Comment by gundalow** [comment]
Needs_revision -  docs

**Comment by privateip** [comment]
ready_for_review

**Pull request merged**